### PR TITLE
Laboratory work #1, Ulyana Nagornaya - 22FPL2 

### PR DESCRIPTION
**Pull request by Ulyana-Nagornaya**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print('Hello, world!')
```
Comment (artyomtugaryov) [inline main.py:1]: Remove the file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Nothing to review :(

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
nothing to review again:(

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (artyomtugaryov) [inline .vscode/settings.json:1]: revert

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -3,14 +3,24 @@
...
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    text = text.lower()
+    tokens = []
+
+    for word in text:
+        for letter in word:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:19]: Do you think you eed this loop?

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -19,6 +29,19 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
+        if not isinstance(token,str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:35]: Why do not you do this check in the main loop  ?

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -28,7 +51,14 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not isinstance(language,str) or not isinstance(text,str):
+        return None
+
+    dict_of_freq = calculate_frequencies(tokenize(text))
 
+    if isinstance(dict_of_freq, dict):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:59]: strange - reverse the condition

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -11,6 +12,16 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    text = text.lower()
+    tokens = []
+
+    for letter in text:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:20]: do the for in one line, please

File: lab_1_classify_by_unigrams/start.py [hunk 7]
```diff
@@ -1,19 +1,45 @@
...
     """
-    with open("assets/texts/en.txt", "r", encoding="utf-8") as file_to_read_en:
-        en_text = file_to_read_en.read()
-    with open("assets/texts/de.txt", "r", encoding="utf-8") as file_to_read_de:
-        de_text = file_to_read_de.read()
+    paths_to_profiles = ['assets/profiles/en.json', 'assets/profiles/es.json',
+                         'assets/profiles/fr.json','assets/profiles/it.json',
+                         'assets/profiles/ru.json','assets/profiles/tr.json']
+
+    # with open("assets/texts/en.txt", "r", encoding="utf-8") as file_to_read_en:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:18]: what is it? revert

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -98,10 +214,23 @@ def detect_language_advanced(unknown_profile: dict[str, str | dict[str, float]],
...
 
 
 def print_report(detections: list[tuple[str, float]]) -> None:
     """
     Prints report for detection of language
     :param detections: a list with distances for each available language
     """
+    for language in detections:
+        print(f'{language[0]}: MSE {language[1]:.5f}')
+
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:236]: strange change

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -19,6 +30,16 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list) or not all(isinstance(token, str) for token in tokens):
+        return None
+
+    dict_of_freq = {}
+    for token in tokens:
+        quantity_of_token = tokens.count(token)
+        frequency = quantity_of_token / len(tokens)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:39]: I believe the world be :
1. Do not calculate `len(tokens)` every iteration
2. Do not calculate the save frequencies for the same tokens many times

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/start.py (hunk 7 above)
Comment (Ulyana-Nagornaya) [reply lab_1_classify_by_unigrams/start.py:18]: What exactly do you mean? Should I delete code with # and remain only code for 10 score?

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (Ulyana-Nagornaya) [reply lab_1_classify_by_unigrams/main.py:236]: What exactly does look strange? As I see I get a list with a tuple and then I print only name of a language and an appropriate MSE score. I can’t understand properly: do I lose something?

**Pull request closed without merging**